### ISSUE TITLE
Remote service module can only call static endpoints. We should add support for dynamic endpoints as well

### ISSUE THREAD

**Issue by YourTechBud** [spaceuptech/space-cloud#492]
### The problem faced currently?
Most http servers have dynamic endpoints. By dynamic i mean path parameters. The remote service module cannot generate those endpoints yet.
### How can we solve it?
Let's allow the user to define the endpoint path as `/v1/users/{args.params.userId}`.

The front end graphql query will be like 
```graphql
{
  endpoint (params:{ userId: "user1"}, body : $body) @service {
    ...the response body comes here
  }
}
```

In this case, `space-cloud` will query the http service on the `/v1/users/user1` endpoint.

Needless to say the `body` field in the request will be passed as the body of the request made to the http service.

@Jayesh333, once you agree with this, we can push this out in `v0.13.1` itself.

>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

**Comment by Jayesh333** [comment]
Yeah! Sounds like an amazing idea! Let's pull this in `v0.13.1`

**Comment by YourTechBud** [comment]
As discussed earlier, we are finally going forward with something like this:
**Endpoint:** /v1/users/{args.userId}
```graphql
{
  endpoint (userId: "user1", param2: SOME_VALUE) @service {
    ...the response body comes here
  }
}
```

Query to the service
**Endpoint:** `/v1/users/user1`
**Body:**
```json
{
  "userId": "user1",
  "param2": SOME_VALUE
}
```

Also, the user can configure the http method to be used

**Comment by YourTechBud** [comment]
Resolved in `v0.13.1`
